pkgs/development/libraries/libcprime/0001-fix-application-dirs.patch

```diff
@@ -0,0 +1,28 @@
+From 2a3948b9a1b99bb1749e2e1c9996ee35a3c18308 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Mustafa=20=C3=87al=C4=B1=C5=9Fkan?= <musfay@protonmail.com>
+Date: Wed, 28 Jul 2021 02:26:39 +0300
+Subject: [PATCH] fix application dirs
+
+---
+ cprime/systemxdg.cpp | 5 +++--
+ 1 file changed, 3 insertions(+), 2 deletions(-)
+
+diff --git a/cprime/systemxdg.cpp b/cprime/systemxdg.cpp
+index f9eee66..2ceb45c 100644
+--- a/cprime/systemxdg.cpp
++++ b/cprime/systemxdg.cpp
+@@ -233,8 +233,9 @@ void SystemXdgMime::setApplicationAsDefault( QString appFileName, QString mimety
+ SystemXdgMime::SystemXdgMime() {
+ 
+ 	appsDirs << QDir::home().filePath( ".local/share/applications/" );
+-	appsDirs << "/usr/local/share/applications/" << "/usr/share/applications/";
+-	appsDirs << "/usr/share/applications/kde4/" << "/usr/share/gnome/applications/";
++	appsDirs << "/run/current-system/sw/share/applications/";
++	appsDirs << "/run/current-system/sw/share/applications/kde4/";
++	appsDirs << "/run/current-system/sw/share/gnome/applications/";
```

Comment:
This wouldn't work on none nixos, right?

Reply:
We can add `QDir::home().filePath( ".nix-profile/share/applications/" );` to support GUI applications installed via Nix but I think we shouldn't import host applications located at `/usr`.

Reply:
I think that might work.

You are also sure kde4 is still right? Aren't we at version 5 now?

Reply:
> I think that might work.
Added to patch

> You are also sure kde4 is still right? Aren't we at version 5 now?
All .desktop files merged to top `share/applications`. Upstream wants to keep it probably for backwards compatibility.